recipes/date/all/conanfile.py

```diff
@@ -1,29 +1,85 @@
     license = "MIT"
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+    exports_sources = ["patches/*", "CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_system_tz_db": [True, False],
+               "use_tz_db_in_dot": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_system_tz_db": True,
+                       "use_tz_db_in_dot": False}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
 
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+    def configure(self):
+        #requires at least c++11 to work.
+        tools.check_min_cppstd(self, "11")
+
+        if self.settings.os == "Windows":
+            del self.options.use_system_tz_db
+            del self.options.use_tz_db_in_dot
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["ENABLE_DATE_TESTING"] = False
+        if not self.settings.os == "Windows":
+            cmake.definitions["USE_TZ_DB_IN_DOT"] = self.options.use_tz_db_in_dot
+            cmake.definitions["USE_SYSTEM_TZ_DB"] = self.options.use_system_tz_db
+
+        cmake.configure()
+        return cmake
+
+    def requirements(self):
+        if self.settings.os == "Windows" or not self.options.use_system_tz_db:
+            self.requires("libcurl/7.67.0")
+
+    def build(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="date.h", dst=os.path.join("include", "date"),
-                  src=os.path.join(self._source_subfolder, "include", "date"))
+        self.copy(pattern="LICENSE.txt", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "CMake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.libs.append("pthread")
+        if self.settings.os == "Windows":
+            self.cpp_info.libs.extend(["ws2_32", "Crypt32"])
```

Comment:
Are you sure these system libraries are direct dependencies of date? Are they not coming from libcurl?

Reply:
Well, this was the reason why I even started doing this change!
but now I see this could be possibly solved by fixing the cmake.. commiting.

Reply:
Thanks - turned out to be not necessary in the end.